cli/tests/buildenv.bats

```diff
@@ -0,0 +1,269 @@
+# bats test_tags=buildenv:runtime
+@test "Verify build closure contains only toplevel packages" {
+  run --separate-stderr "$BUILDENV_BIN" \
+    "$INPUT_DATA/buildenv/lockfiles/build/manifest.lock"
+  assert_success
+  assert_equal "${#lines[@]}" 1 # 1 result
+  runtime_outPath=$(echo "${lines[0]}" | jq -er '.[0] | .outputs.runtime')
+  develop_outPath=$(echo "${lines[0]}" | jq -er '.[0] | .outputs.develop')
+  build_myhello_outPath=$(echo "${lines[0]}" | jq -er '.[0] | .outputs."build-myhello"')
+  assert test -x "${runtime_outPath}/bin/hello"
+  assert test -x "${develop_outPath}/bin/hello"
+  assert test -x "${build_myhello_outPath}/bin/hello"
+  assert test -x "${runtime_outPath}/bin/coreutils"
+  assert test -x "${develop_outPath}/bin/coreutils"
+  assert test -x "${build_myhello_outPath}/bin/coreutils"
+  assert test -x "${runtime_outPath}/bin/vim"
+  assert test -x "${develop_outPath}/bin/vim"
+  assert test ! -e "${build_myhello_outPath}/bin/vim"
+}
+
+# bats test_tags=buildenv:runtime,buildenv:runtime-packages-only-hello
+@test "Verify build closure contains only hello with runtime-packages attribute" {
+  run --separate-stderr "$BUILDENV_BIN" \
+    "$INPUT_DATA/buildenv/lockfiles/build-runtime-packages-only-hello/manifest.lock"
+  assert_success
+  assert_equal "${#lines[@]}" 1 # 1 result
+  runtime_outPath=$(echo "${lines[0]}" | jq -er '.[0] | .outputs.runtime')
+  develop_outPath=$(echo "${lines[0]}" | jq -er '.[0] | .outputs.develop')
+  build_myhello_outPath=$(echo "${lines[0]}" | jq -er '.[0] | .outputs."build-myhello"')
+  assert test -x "${runtime_outPath}/bin/hello"
+  assert test -x "${develop_outPath}/bin/hello"
+  assert test -x "${build_myhello_outPath}/bin/hello"
+  assert test -x "${runtime_outPath}/bin/coreutils"
+  assert test -x "${develop_outPath}/bin/coreutils"
+  assert test ! -e "${build_myhello_outPath}/bin/coreutils"
+  assert test -x "${runtime_outPath}/bin/vim"
+  assert test -x "${develop_outPath}/bin/vim"
+  assert test ! -e "${build_myhello_outPath}/bin/vim"
+}
+
+# bats test_tags=buildenv:runtime,buildenv:runtime-packages-not-toplevel
+@test "Verify build closure can only select toplevel packages from runtime-packages attribute" {
+  run "$BUILDENV_BIN" \
+    "$INPUT_DATA/buildenv/lockfiles/build-runtime-packages-not-toplevel/manifest.lock"
+  assert_failure
+  assert_output --regexp "error: package 'vim' is not in 'toplevel' pkg-group"
+}
+
+# bats test_tags=buildenv:runtime,buildenv:runtime-packages-not-found
+@test "Verify build closure cannot select nonexistent packages in runtime-packages attribute" {
+  run "$BUILDENV_BIN" \
+    "$INPUT_DATA/buildenv/lockfiles/build-runtime-packages-not-found/manifest.lock"
+  assert_failure
+  assert_output --regexp "error: package 'goodbye' not found in '\[install\]' section of manifest"
+}
+
+# ---------------------------------------------------------------------------- #
+#
+#
+#
+# ============================================================================ #
```

Comment:
note nonblocking: might make more sense to move these to Rust https://github.com/flox/flox/issues/2207

Reply:
I think breaking the builds out as a `provider` of sorts would let us test these builds in a similar way as package builds.
I agree this is non-blocking for this PR which does not introduce any rust side changes.

NB. adding these comparatively heavy tests made the unit tests considerably slower -- it might be worth configuring those as separate CI jobs.

Reply:
Matthew explained that Yannik's comment was in response to his comment, so as far as I'm aware this isn't a blocker for merging 2226 and I'll go ahead and resolve the conversation. I agree it may make sense to move tests to rust. FTR once the required packages are present in the store the entirety of `buildenv.bats` only takes a second to run on my machine.